### DISCUSSION TITLE
Lesson 11 - [FAIL. Reason: cannot `prank` for a broadcasted transaction;]

### DISCUSSION THREAD

**Discussion by DeanKrupka** [Cyfrin/foundry-full-course-cu#1279] [Q&A]
I am getting an error originating from my use of vm.prank(USER) for the first time. I really am confounded as I was testing and had things functioning within these contracts days ago and have not changed much since then.  Wondering if anyone has encountered this error and knows a remedy:

```
[FAIL. Reason: cannot `prank` for a broadcasted transaction; pass the desired `tx.origin` into the `broadcast` cheatcode call] 
```

The function the error is coming from is:
```
function testFlipTokenToSad() public {
        vm.prank(USER);
        moodNft.mintNft();

        vm.prank(USER);
        moodNft.flipMood(0);

        assertEq(keccak256(abi.encodePacked(moodNft.tokenURI(0))), keccak256(abi.encodePacked(SAD_MOOD_URI)));
    }
```
^This function is from the MoodNftIntegrations.t.sol contract in Lesson 11.

**Comment by xyz899** [comment]
Hi, 
Once a transaction has been broadcasted, you can't use prank to alter its sender or origin. For this issue, ensure that you're not trying  to use prank after a transaction has been broadcasted. If you need to simulate a certain sender or origin for a broadcasted transaction, you should do so before calling broadcast

**Reply by DeanKrupka** [reply]
Yes! The issue was related to this. I had a startBroadcast() in one of the Deployers that did not have a corresponding stopBroadcast().
